File: src/positioning/matlab/algorithm/InterpPos.m
function [InterpPosition] = InterpPos(Px,Py,s_time)
Px = Px-s_time;
Pm = [Px.^2 Px ones(size(Px))];

A = inv(Pm'*Pm)*Pm'*Py;
InterpPosition = A(3);


% Py = pm x A
% tr(pm)xPy = tr(Pm)pm x A
% inv(tr(Pm)Pm) x tr(pm)xPy = A

File: src/positioning/keti_ws/test/InterpPos.py
import numpy as np

def InterpPos(Px, Py, s_time):
    Px = Px - s_time
    Pm = np.column_stack((Px**2, Px, np.ones(len(Px))))

    A = np.linalg.inv(Pm.T @ Pm) @ Pm.T @ Py
    InterpPosition = A[2]

    return InterpPosition
